### PR TITLE
Milestone : Add relationships

### PR DESCRIPTION
**Pull request by ahmed025566**
In this milestone you could see that I have done the following:

- I have created classroom class with a method to add students.
- I have created book class to initialize new books with a property rentals in it.
- I have created rentals class to initialize a new rental witch will be stored in rentals in both book and person.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,9 +1,8 @@
 require_relative 'person'
 
 class Student < Person
```
Comment (tugsem) [inline student.rb:3]: - [ ] As a student belongs to a class, please write a method to serve this purpose. Please feel free to check this link for more detail. https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
```
Comment (tugsem) [inline book.rb:8]: - [ ] Please add a method for adding a rental as Book and Person class has-many-through relationship using the Rental class. The book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship

File: person.rb [hunk 3]
```diff
@@ -9,6 +10,7 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @age = age
     @name = name
     @parent_permission = parent_permission
+    @rentals = []
```
Comment (tugsem) [inline person.rb:13]: - [ ] Please also add here a method to add rentals using the Person class. The person himself should be sent as a parameter to create the new rental achieving the 'has-many' association.

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello @ahmed025566 👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️classes created successfully

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb [hunk 4]
```diff
@@ -1,12 +1,17 @@
...
+    super(age, name, parent_permission: parent_permission)
   end
 
   def play_hooky
     '¯\(ツ)/¯'
   end
+
+  def add_student
+    classroom.add_student(self) unless classroom.students.include?(self)
+  end
```
Comment (tresorsawasawa) [inline student.rb]: - [ ] Good job trying to fix change(s) mentioned by the previous reviewer. However, as you are supposed to add a method that serves a student belongs to a classroom, kindly ensure that you update it to something similar to this one:
```ruby
  attr_reader :classroom

 def initialize(age, classroom, name = 'Unknown', parent_permission: true)
    super(age, parent_permission: parent_permission, name: name)
    @classroom = classroom
 end
  .
  .
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUEST CHANGES :recycle::recycle: 

Hi @ahmed025566 
Great work on making the changes requested by the previous reviewer  👏👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @ahmed025566, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Changes By Previous Reviewers :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**